### PR TITLE
fix: AST rewriting when inline script is very long

### LINKED ISSUE #8043: AST rewriting not adding closing script tag when string is too long
### Current behavior:
The following test case, with `experimentalSourceRewriting` turned on, will not properly render the page. (It doesn't properly render the page without it either, see #1068)

```javascript
  it('can open office online', () => {
    cy.visit('https://www.office.com/launch/excel?ms.officeurl=webapps&auth=2');

    cy.contains('Sign in').should('be', 'visible');
  });
```

It looks like when the source is being rewritten, something is causing the closing `</script>` tag to be omitted. When this tag is omitted, the rest of the page goes missing as well.

Here's what the rewritten response looks like:

Screen Shot 2020-07-20 at 6 32 38 PM

Here's the actual response:

Screen Shot 2020-07-20 at 6 32 58 PM

From some initial investigations it appears that this is a bug caused by the length of the script.

┆Issue is synchronized with this [Jira Bug](https://cypress-io.atlassian.net/browse/TR-156) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by panzarino**
- Closes #8043

### User facing changelog

Fixes an issue where AST rewriting would return an output before the body was done being written. This would happen when the response body was too large and the response would be sent while the body was still being modified.

### Additional details

This following test case would fail with the previous implementation, but now passes: https://github.com/cypress-io/cypress/pull/8048/files#diff-2aeeae49664314a9a544ffdea59ec137R55.

From the issue, here's an example of a body getting cut off:

Screen Shot 2020-07-20 at 6 32 38 PM

Here's what the response looks like now:

Screen Shot 2020-07-21 at 5 19 15 PM

Note: Thanks to Murphy's law, the test case from this issue still does not pass (due to something unrelated).

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub?

### CODE REVIEW

File: packages/rewriter/lib/threads/worker.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ parentPort!.on('message', (req: RewriteRequest) => {
   }
 
   try {
-    const output = _getOutput()
+    const output = await _getOutput()
```
Comment (flotwig) [inline packages/rewriter/lib/threads/worker.ts:61]: @panzarino i thought this wasn't working like we expected, did you figure it out?

Comment (panzarino) [reply packages/rewriter/lib/threads/worker.ts:61]: It was working, turns out the issue I was seeing was from somewhere else and it was completely unrelated. I also tested the async functions specifically and it was working fine.

**Comment by panzarino** [comment]
@flotwig it's split by the `RewritingStream`. Each chunk is the result of the response in `html-rules.ts`, for example when it sends a `rewriter.emitRaw(raw)`, that's a chunk.

**Comment by flotwig** [comment]
@panzarino That doesn't explain why this only happens after a certain body size, since the other test cases emit as one chunk.

**Comment by panzarino** [comment]
@flotwig Wasn't aware of that, not sure why its not working like that this time then.

Should I be concerned with the failing server performance test?

**Comment by flotwig** [comment]
I think that's flake.

**Pull request merged**